File: common/src/main/java/dev/mayaqq/estrogen/networking/messages/s2c/ThighHighStylesPacket.java
package dev.mayaqq.estrogen.networking.messages.s2c;

import com.teamresourceful.bytecodecs.base.object.ObjectByteCodec;
import com.teamresourceful.resourcefullib.common.bytecodecs.ExtraByteCodecs;
import com.teamresourceful.resourcefullib.common.network.Packet;
import com.teamresourceful.resourcefullib.common.network.base.ClientboundPacketType;
import com.teamresourceful.resourcefullib.common.network.base.PacketType;
import com.teamresourceful.resourcefullib.common.network.defaults.CodecPacketType;
import dev.mayaqq.estrogen.Estrogen;
import dev.mayaqq.estrogen.registry.EstrogenItems;
import net.minecraft.resources.ResourceLocation;

import java.util.List;

public record ThighHighStylesPacket(List<ResourceLocation> styles) implements Packet<ThighHighStylesPacket> {

    public static final ClientboundPacketType<ThighHighStylesPacket> TYPE = new Type();

    @Override
    public PacketType<ThighHighStylesPacket> type() {
        return TYPE;
    }

    private static class Type extends CodecPacketType.Client<ThighHighStylesPacket> {

        public Type() {
            super(
                ThighHighStylesPacket.class,
                Estrogen.id("thigh_high_styles"),
                ObjectByteCodec.create(
                    ExtraByteCodecs.RESOURCE_LOCATION.listOf().fieldOf(ThighHighStylesPacket::styles),
                    ThighHighStylesPacket::new
                )
            );
        }

        @Override
        public Runnable handle(ThighHighStylesPacket message) {
            return () -> EstrogenItems.THIGH_HIGHS.get().loadStyles(message.styles());
        }
    }
}

File: src/main/kotlin/dev/mayaqq/estrogen/network/messages/s2c/ThighHighStylesPacket.kt
package dev.mayaqq.estrogen.network.messages.s2c

import dev.mayaqq.estrogen.content.EstrogenItems
import invoke.kitty.kritter.serialization.builtins.ResourceLocationSerializer
import kotlinx.serialization.Serializable
import net.minecraft.resources.ResourceLocation

@Serializable @JvmRecord
data class ThighHighStylesPacket(val styles: List<@Serializable(ResourceLocationSerializer::class) ResourceLocation>) {

//            Rek;;;;;;;;;;;;\
//        puppys contribution to estrogen c:
//        [[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[[ourceLocationByteCodec.listOf()

    fun handle() {
        EstrogenItems.ThighHighs.value!!.loadStyles(styles)
    }
}
